src/types.ts

```diff
@@ -1,4 +1,4 @@
-export interface ConfigInterface {
+export interface ConfigInterface<Data = any, Error = any> {
```

Comment:
Wasnt sure if `Error` should default to `any` or `Error`

Reply:
Added defaults to avoid a BC break

Reply:
Yeah, I think it's good to leave the possibility of throwing anything. There might be some interesting use cases.